extras/check_code_style.sh

```diff
@@ -0,0 +1,215 @@
+echo "vera++ version: `vera++ --version`"
+
+# check cppcheck 1.69 is installed correctly
+$CPPCHECK --enable=all --inconclusive --std=c++03 ./nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $CPPCHECK for cppcheck is not working!"
+cppcheck_version=`$CPPCHECK --version | sed 's/^Cppcheck //'`
+echo "cppcheck version: $cppcheck_version"
+if [[ "x$cppcheck_version" != "x1.69" ]]; then
+  bail_out "Require cppcheck version 1.69. Not $cppcheck_version ."
+fi
+
+# clang-format is installed correctly
+$CLANG_FORMAT -style=./.clang-format ./nest/main.cpp >/dev/null 2>&1 || usage 1 "Executable $CLANG_FORMAT for clang-format is not working!"
+clang_format_version=`$CLANG_FORMAT --version | sed -${EXTENDED_REGEX_PARAM} 's/^.*([0-9]\.[0-9])\..*/\1/'`
+echo "clang-format version: $clang_format_version"
+if [[ "x$clang_format_version" != "x3.6" ]]; then
+  bail_out "Require clang-format version 3.6. Not $clang_format_version ."
+fi
+
+# Extracting changed files between two commits
+if [[ "x$ONLY_FILE" != "x" && -f $ONLY_FILE ]]; then
+  file_names=$ONLY_FILE
+else
+  file_names=`git diff --name-only $START_SHA..$END_SHA`
+fi
+format_error_files=""
+
+echo
+echo "Performing static analysis on:"
+for f in $file_names; do
+  echo "  $f"
+done
+if $INCREMENTAL; then
+  echo "Press enter to continue."
+  read
+else
+  echo
+fi
+
+for f in $file_names; do
+  if [ ! -f $f ]; then
+    echo "$f : Is not a file or does not exist anymore."
+    continue
+  fi
+  # filter files
+  case $f in
+    *.h | *.c | *.cc | *.hpp | *.cpp )
+      echo
+      echo "Static analysis on file $f:"
+      if $INCREMENTAL; then
+        echo "Press enter to continue."
+        read
+      else
+        echo
+      fi
+
+      echo " - clang-format for $f: (critical: if there are diffs, perform $CLANG_FORMAT -i $f )"
+      # clang format creates tempory formatted file
+      $CLANG_FORMAT $f > ${f}_formatted.txt
+      # compare the committed file and formatted file and 
+      # writes the differences to a temp file
+      diff $f ${f}_formatted.txt | tee ${f}_clang_format.txt
+
+      if [ -s ${f}_clang_format.txt ]; then 
+        # file exists and has size greater than zero
+        format_error_files="$format_error_files $f"
+      fi
+
+      # remove temporary files
+      rm ${f}_formatted.txt
+      rm ${f}_clang_format.txt
+
+      # Vera++ checks the specified list of rules given in the profile 
+      # nest which is placed in the <vera++ root>/lib/vera++/profile
+      echo " - vera++ for $f: (please consider fixing all warnings emitted here.)"
+      $VERA --profile nest $f
+
+      # perform other static analysis
+      echo " - cppcheck for $f: (suggestions for improvements)"
+      $CPPCHECK --enable=all --inconclusive --std=c++03 $f
+
+      ;;
+    *)
+      echo "$f : not a C/CPP file. Do not do static analysis / formatting checking."
```

Comment:
"Not executing static analysis / formatting checking." instead of "Do not do ..."